apps/examples/smartfs/buffer_optimization/Kconfig_ENTRY

```diff
@@ -0,0 +1,3 @@
+config ENTRY_SMARTFS_BUFFER_OPTIMIZATION
```

Comment:
How about changing `ENTRY_SMARTFS_BUFFER_OPTIMIZATION` to `ENTRY_SMARTFS_BUF_OPTIMIZATION`?
Another configs are matched with the original config name, but this config is not.